packages/react-reconciler/src/ReactFiberHooks.new.js

```diff
@@ -182,8 +181,19 @@ type StoreConsistencyCheck<T> = {
   getSnapshot: () => T,
 };
 
+type EventFunctionState<T> = {
+  event: EventFunctionWrapper<T>,
+  nextEvent: EventFunction<T>,
+};
+type EventFunctionWrapper<T> = {
+  (): T,
+  _current: EventFunction<T>,
+};
+type EventFunction<T> = () => T;
```

Comment:
Didn't notice this in the original PR, but this type should also enforce that the inputs args to the wrapped function are the same as the original function.

Not sure the best way to model this in Flow. This seems to work but I imagine there's something better: https://flow.org/try/#0PQKgBAAgZgNg9gdzCYAoVUCuA7AxgFwEs5sxMBnAUwFEA3S7fAHgEEAnAc3IBowAlSvkxtsvAGIAuMAAoAdPPZsAhgE9WncgD4AlGAC8m-oOHZN0qNiljtU6XIVtla9lx37DAoSN0BvVGDA2YxEZeVklDV0DMD8AuMDg0gt7cMiAbn8wAF8MrKA

It doesn't matter so much for our internal types because we don't export them, but it's happened before that people have look in our codebase and used our internal types as a reference when adding public ones.